src/bson.rs

```diff
@@ -1041,7 +1041,12 @@ impl From<crate::DateTime> for chrono::DateTime<Utc> {
 
 impl<T: chrono::TimeZone> From<chrono::DateTime<T>> for crate::DateTime {
     fn from(x: chrono::DateTime<T>) -> Self {
-        DateTime(x.with_timezone(&Utc))
+        let dt = x.with_timezone(&Utc);
+
+        DateTime(chrono::DateTime::<Utc>::from_utc(
```

Comment:
Now that we're ensuring all `bson::DateTime`s are millisecond precision, can we remove the `ser::Error::SubMillisecondPrecisionDateTime` error case and associated error checking logic during BSON serialization?

Reply:
Yes, I have removed it.